src/python/dart/util/dataset_guess.py

```diff
@@ -22,12 +22,18 @@ def infer_dataset_data(s3_path, max_lines):
     :type max_lines: int
     :param max_lines: The maximum number of lines to peek.
     """
-    guesses, has_header, compression = get_guesses(s3_path, max_lines)
+    guesses, has_header, compression, dialect = get_guesses(s3_path, max_lines)
     columns = columns_with_best_guess(guesses, has_header)
     location = guess_location(s3_path)
     return Dataset(data=DatasetData(name='guessed_dataset',
                                     table_name='public',
-                                    data_format=DataFormat(file_format=FileFormat.TEXTFILE, row_format=RowFormat.DELIMITED),
+                                    data_format=DataFormat(
+                                        file_format=FileFormat.TEXTFILE,
+                                        row_format=RowFormat.DELIMITED,
+                                        delimited_by=dialect.delimiter,
```

Comment:
If the sniffer can't tell from the sample what a value should be then it's set to None, which is what it defaulted to when creating DataFormat.